merge/list_test.go

```diff
@@ -0,0 +1,282 @@
+package merge_test
+
+import (
+	"testing"
+
+	"sigs.k8s.io/structured-merge-diff/typed"
+
+	"sigs.k8s.io/structured-merge-diff/merge"
+)
+
+func TestInlineListMerge(t *testing.T) {
+	parser, err := typed.NewParser(`types:
+- name: lists
+  struct:
+    fields:
+    - name: items
+      type:
+        list:
+          elementType:
+            scalar: string
+          elementRelationship: associative`)
+	if err != nil {
+		t.Fatalf("Failed to create parser: %v", err)
+	}
+
+	t.Run("disjointItemsOrder", func(t *testing.T) {
```

Comment:
I'd write a different test rather than abusing the `t.Run` thing maybe.

Reply:
I thought that's what its meant for as it allows grouping and produces the according output:
```
--- FAIL: TestInlineListMerge (0.01s)
    --- FAIL: TestInlineListMerge/orderConstraintCheck (0.00s)
        /home/kwiesmueller/go/src/sigs.k8s.io/structured-merge-diff/merge/list_test.go:272: Wanted err = <nil>, got .items[=true]: expected string, got true
```

Reply:
I mostly use these for table-driven tests, but sure why not